backend/trafficrules/src/main/java/com/coffeejug/trafficrules/service/UserService.java

```diff
@@ -0,0 +1,70 @@
+package com.coffeejug.trafficrules.service;
+
+import com.coffeejug.trafficrules.db.Progress;
+import com.coffeejug.trafficrules.db.User;
+import com.coffeejug.trafficrules.dto.UserDto;
+import com.coffeejug.trafficrules.projection.ProgressProjection;
+import com.coffeejug.trafficrules.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class UserService {
+
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private ProgressService progressService;
+
+
+    public UserDto createNew() {
+
+        User user = new User();
+        user = userRepository.save(user);
+        UserDto userDto = new UserDto(user.getUuid().toString());
+        return userDto;
+    }
+
+    public List<ProgressProjection> getUserProgress(String uuid) {
+
+        if (goodUUID(uuid)) {
+            User user = userRepository.findById(UUID.fromString(uuid)).get();
+            return progressService.findAllByUser(user);
+        }
+        return new ArrayList<>();
+    }
+
+    private boolean goodUUID(String uuid) {
+
+        if ((uuid != null) && (uuid.length() == 36)) {
+            return true;
+        }
+        return false;
+    }
+
+
+    public User getOne(UUID id) {
+
+        if (userRepository.existsById(id)) {
+            return userRepository.getOne(id);
+        }
+        return null;

```

Comment:
What if we will use it from another Service, not Controller ?

Reply:
If this method could return empty result then let's use `Optional`. Moreover, it's better to use the repository method that returns `Optional`, it's called `findOne` instead of making to calls to DB

Reply:
findOne() uses "select *", but  existsById() uses "select count()" which is faster. Result of this optimization depends of which requests more: with write or wrong code

Reply:
moreover: In my case you need to check if value is null in yours you check Optional isPresent()

Reply:
Premature optimization is the root of all evil. We don't have any performance issue, so let's not sacrifice the code quality for resolving something that is not a problem for us :)
Regarding the null checking, Optional has a perfect method called `ifPresent` that allows us to work in a declarative style, as `if(optional.isPresent())` is considered a bad practice